src/writesbml.jl

```diff
@@ -0,0 +1,86 @@
+        isnothing(compartment.constant) || ccall(sbml(:Compartment_setConstant), Cint, (VPtr, Cint), compartment_t, Cint(compartment.constant))
+        isnothing(compartment.spatial_dimensions) || ccall(sbml(:Compartment_setSpatialDimensions), Cint, (VPtr, Cuint), compartment_t, Cuint(compartment.spatial_dimensions))
+        isnothing(compartment.size) || ccall(sbml(:Compartment_setSize), Cint, (VPtr, Cdouble), compartment_t, Cdouble(compartment.size))
+        isnothing(compartment.units) || ccall(sbml(:Compartment_setUnits), Cint, (VPtr, Cstring), compartment_t, compartment.units)
+        isnothing(compartment.notes) || ccall(sbml(:SBase_setNotesString), Cint, (VPtr, Cstring), compartment_t, compartment.notes)
+        isnothing(compartment.annotation) || ccall(sbml(:SBase_setAnnotationString), Cint, (VPtr, Cstring), compartment_t, compartment.annotation)
+        res = ccall(sbml(:Model_addCompartment), Cint, (VPtr, VPtr), model, compartment_t)
+        !iszero(res) && @warn "Failed to add compartment \"$(id)\": $(OPERATION_RETURN_VALUES[res])"
+    end
+
+    # Add gene products
+    for (id, gene_product) in mdl.gene_products
+        geneproduct_t = ccall(sbml(:GeneProduct_create), VPtr, (Cuint, Cuint, Cuint), 3, 2, 2) # TODO: check what the PkgVersion should be
+        ccall(sbml(:GeneProduct_setId), Cint, (VPtr, Cstring), geneproduct_t, id)
+        isnothing(gene_product.name) || ccall(sbml(:GeneProduct_setName), Cint, (VPtr, Cstring), geneproduct_t, gene_product.name)
+        isnothing(gene_product.label) || ccall(sbml(:GeneProduct_setLabel), Cint, (VPtr, Cstring), geneproduct_t, gene_product.label)
+        isnothing(gene_product.notes) || ccall(sbml(:SBase_setNotesString), Cint, (VPtr, Cstring), geneproduct_t, gene_product.notes)
+        isnothing(gene_product.annotation) || ccall(sbml(:SBase_setAnnotationString), Cint, (VPtr, Cstring), geneproduct_t, gene_product.annotation)
+        # TODO: add the gene product to the FBC package
+        # res = ccall(sbml(:FbcModelPlugin_addGeneProduct), Cint, (VPtr, VPtr), model_fbc, geneproduct_t)
+        # !iszero(res) && @warn "Failed to add gene product \"$(id)\": $(OPERATION_RETURN_VALUES[res])"
+    end
+
+    # Add species
+    for (id, species) in mdl.species
+        species_t = ccall(sbml(:Species_create), VPtr, (Cuint, Cuint), 3, 2)
+        ccall(sbml(:Species_setId), Cint, (VPtr, Cstring), species_t, id)
+        isnothing(species.name) || ccall(sbml(:Species_setName), Cint, (VPtr, Cstring), species_t, species.name)
+        isnothing(species.compartment) || ccall(sbml(:Species_setCompartment), Cint, (VPtr, Cstring), species_t, species.compartment)
+        isnothing(species.boundary_condition) || ccall(sbml(:Species_setBoundaryCondition), Cint, (VPtr, Cint), species_t, species.boundary_condition)
+        # isnothing(species.formula) || ccall(sbml(:Species_setFormula), Cint, (VPtr, Cstring), species_t, species.compartment)
```

Comment:
I fear this is related to the problem with gene products: how do we write the formula?  There is no `Species_setFormula` function, and we read from the FBC package: https://github.com/LCSB-BioCore/SBML.jl/blob/8ee591a473a7ecb911c4dff474ca1e8db7337e86/src/readsbml.jl#L295

Reply:
(yeah looks like there's no doxygen around the source)

Reply:
Ok, that's useful to know, thanks!  But I think we still don't know how to create the plugin